src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -317,9 +317,18 @@ def _dispatchWflow(self, wflow):
             #       {REPLICATING, STUCK, SUSPENDED, WAITING_APPROVAL}.)
             #       We still need some proper logging for them.
             msg = "Skipping workflow: %s - tape transfers are not yet completed." % wflow['RequestName']
-            msg += " Will retry again in the next cycle."
+            msg += " Will retry again in the next cycle. deleting ONLY the input MS-Transferor rules as wf is announced."
             self.logger.info(msg)
             self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
+            for pline in self.mstrlines:
+                try: 
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline while cleaning input MS-Transferor rules. Workflow: %s. Error:  \n%s. " % (pline.name, wflow['RequestName'], str(ex))
```

Comment:
Please break this into shorter (~<100 chars) lines. And use MSTransferor instead of MS-Transferor, just for the sake of consistency.